### PR TITLE
Add Canary Deployment tests

### PR DESCRIPTION
**Pull request by Samze**
Add Baras tests for canary deployments

Relates to https://github.com/cloudfoundry/cloud_controller_ng/issues/3837

### CODE REVIEW

File: baras/deployments.go [hunk 1]
```diff
@@ -170,9 +172,194 @@ var _ = Describe("deployments", func() {
...
 
 		It("completes the deployment", func() {
-			deploymentGUID := CreateDeployment(appGUID)
+			deploymentGUID := CreateDeployment(appGUID, "rolling")
 			Expect(deploymentGUID).ToNot(BeEmpty())
 			WaitUntilDeploymentReachesStatus(deploymentGUID, "FINALIZED", "DEPLOYED")
 		})
 	})
+
+	Describe("Canary deployments", func() {
```
Comment (a-b) [inline baras/deployments.go:181]: thought(non-blocking): Should we have similar describe block for rolling deployments?

**Review by joaopapereira — APPROVED** [review]
LGTM

**Review by a-b — APPROVED** [review]
LGTM

File: baras/deployments.go (hunk 1 above)
Comment (Samze) [reply baras/deployments.go:181]: Some of the existing tests are not specific for rolling, so we'd need to do some surgery to unpick the strategy specific tests. For now I think we can leave it and revisit it if we add a future strategy.

**Review by xandroc — APPROVED** [review]
nice work!

**Review by tcdowney — APPROVED** [review]
LGTM

**Pull request merged**